src/plugins/hidsim/hidsim.h

```diff
@@ -131,10 +133,25 @@ class hidsim : public plugin
     ~hidsim();
 
 private:
-    std::thread t;
+    /* Configuration passed via CLI */
     std::string sock_path;
     std::string template_path;
-    sig_atomic_t has_to_stop;
+    std::string win32k_json_path;
+
+    bool is_monitor;
+    bool is_gui_support;
+
+    /* Worker threads */
+    std::thread thread_inject;
+    std::thread thread_reconstruct;
+
+    /* Thread communication */
+    volatile sig_atomic_t has_to_stop;
+    volatile sig_atomic_t coords;
```

Comment:
`sig_atomic_t` is not a safe way of sharing data between threads. Please use appropriate constructs from C++ standard library.

Reply:
Changed in 6ef93dc6bc9e3b9ec0d32272d06eac5fe6b1a656 and df16b512c921ca31a2715fb343bcb767609df77c to use C++'s `std::atomic<...>` 

(Although I would have found it nicer to use the `atomic_*` primitives specified in C11)